design-documents/0006-metadata-api/text.md

```diff
@@ -0,0 +1,118 @@
+
+When updating a public dataset (a dataset that belongs to a public collection, with publicized urls for viewing the datasets via the Explorer), it is currently necessary to overwrite the dataset in the S3 bucket in order to maintain the explorer urls, rather than simply redirecting to the new dataset location.
+
+This also complicates the process of deleting a dataset that has been published, while still communicating any relevant information to the end user. If the dataset is simply deleted from S3, the explorer will have no knowledge of it, and when someone attempts to navigate to the Explorer page for the deleted dataset, it can only display a 404, after looking for the cxg in the s3 bucket and finding nothing there.
+
+Additionally cxg's are more or less read-only. But the Portal needs to communicate new information about the dataset after it writes the cxg. For example, perhaps there should be an update to the title of the dataset. There is no way to do this (and communicate the change to the Explorer) without rewriting the entire cxg.
+
+While newly created datasets are displayed in the Explorer under a url referencing their dataset uuid, historically created (but still supported) urls identify the datasets via the dataset names, further complicating references to specific datasets.
+
+## Product Requirements
+
+[Data consumers can easily discover the portal and other collections from cellxgene explorer](https://app.zenhub.com/workspaces/single-cell-5e2a191dad828d52cc78b028/issues/chanzuckerberg/single-cell/169)
+
+[Data consumers can easily discover and review metadata for a dataset in cellxgene explorer](https://app.zenhub.com/workspaces/single-cell-5e2a191dad828d52cc78b028/issues/chanzuckerberg/single-cell/171)
+
+[Curators may privately revise a published collection](https://app.zenhub.com/workspaces/single-cell-5e2a191dad828d52cc78b028/issues/chanzuckerberg/single-cell/36) with emphasis on [Curators may update published datasets in the private revision of a published collection](https://app.zenhub.com/workspaces/single-cell-5e2a191dad828d52cc78b028/issues/chanzuckerberg/single-cell/141) and [Curators may delete published datasets in a private revision for a public collection](https://app.zenhub.com/workspaces/single-cell-5e2a191dad828d52cc78b028/issues/chanzuckerberg/single-cell/138)
+
+## Detailed Design | Architecture | Implementation
+
+Summary:
+Create an alias tier between the Explorer and the Data Portal to allow for more flexibility in linkage between a url and a dataset. The explorer will use GET `datasets/meta?url={cellxgene_url}` to retrieve the collection id for the dataset as well as a link to where in S3 the dataset to display is stored. This will require changes in both the portal and the explorer code base.
```

Comment:
States _collection id_ here, but returns a _collection url_ in the **Response** below?

Reply:
typo -- fixed